From 60756da2913fdfa36388c5ee5d149e932747157f Mon Sep 17 00:00:00 2001
From: Jacob McSwain <jacob@mcswain.dev>
Date: Sun, 2 Jul 2023 04:22:36 -0500
Subject: [PATCH 15/26] Improved bad route handling

---
 src/linux/kernel_routes_nl.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/linux/kernel_routes_nl.c b/src/linux/kernel_routes_nl.c
index 22ed9f59..8e5235e3 100644
--- a/src/linux/kernel_routes_nl.c
+++ b/src/linux/kernel_routes_nl.c
@@ -464,6 +464,14 @@ int olsr_new_netlink_route(unsigned char family, uint32_t rttable, unsigned int
     if (!set && errno == 11 && if_ifwithindex(if_index) == NULL) {
       err = 0;
     }
+    /* Failed to delete a route - ignore the error so the route will be successfully remove from internal state */
+    if (!set && (errno == 11 || errno == 122)) {
+      err = 0;
+    }
+    /* Failed to add a route - ignore the error to avoid this happening continually */
+    else if (set && (errno == 11 || errno == 122)) {
+      err = 0;
+    }
   }
 
   return err;
-- 
2.41.0

