From 7242bf6bd043e7543504ecce6b74b5670efe8aa3 Mon Sep 17 00:00:00 2001
From: Tim Wilkinson <tim.j.wilkinson@gmail.com>
Date: Sun, 1 Aug 2021 14:56:25 -0700
Subject: [PATCH 22/26] Isolation fix for hello messages

---
 src/packet.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/src/packet.c b/src/packet.c
index 3f8c9ae5..c7b68fda 100644
--- a/src/packet.c
+++ b/src/packet.c
@@ -202,9 +202,15 @@ olsr_build_hello_packet(struct hello_message *message, struct interface_olsr *ou
   OLSR_PRINTF(5, "Not on link:\n");
 #endif /* DEBUG */
 
-  /* Add the rest of the neighbors if running on multiple interfaces and this interface is not isolated */
+  /* If interface is isolated we don't include any other interface neighbors */
 
-  if (ifnet != NULL && ifnet->int_next != NULL && ifnet->mode != IF_MODE_ISOLATED)
+  if (outif->mode == IF_MODE_ISOLATED) {
+    return 0;
+  }
+
+  /* Add the rest of the neighbors if running on multiple interfaces */
+
+  if (ifnet != NULL && ifnet->int_next != NULL)
     OLSR_FOR_ALL_NBR_ENTRIES(neighbor) {
 
 #ifdef DEBUG
-- 
2.41.0

