From 016d5dea1bf10260acf08d642594db49361af145 Mon Sep 17 00:00:00 2001
From: Tim Wilkinson <tim.j.wilkinson@gmail.com>
Date: Sat, 7 Aug 2021 14:27:32 -0700
Subject: [PATCH 24/26] Prevent isolated names flowing into the unisolated
 network

---
 lib/nameservice/src/nameservice.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/lib/nameservice/src/nameservice.c b/lib/nameservice/src/nameservice.c
index f8738431..3d080d92 100644
--- a/lib/nameservice/src/nameservice.c
+++ b/lib/nameservice/src/nameservice.c
@@ -766,6 +766,11 @@ olsr_parser(union olsr_message *m, struct interface_olsr *in_if __attribute__ ((
 
   update_name_entry(&originator, namemessage, size, vtime);
 
+  /* Never forward these messages from an isolated interface */
+  if (in_if->mode == IF_MODE_ISOLATED) {
+    return false;
+  }
+
   /* Forward the message */
   return true;
 }
-- 
2.41.0

