From 8354737075c360338d8f28feb50059f3b9c97471 Mon Sep 17 00:00:00 2001
From: Tim Wilkinson <tim.j.wilkinson@gmail.com>
Date: Tue, 26 Apr 2022 23:20:54 -0700
Subject: [PATCH 26/26] Allow more nameservice entries

---
 lib/nameservice/src/nameservice.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/lib/nameservice/src/nameservice.c b/lib/nameservice/src/nameservice.c
index 3d080d92..f8738431 100644
--- a/lib/nameservice/src/nameservice.c
+++ b/lib/nameservice/src/nameservice.c
@@ -766,11 +766,6 @@ olsr_parser(union olsr_message *m, struct interface_olsr *in_if __attribute__ ((
 
   update_name_entry(&originator, namemessage, size, vtime);
 
-  /* Never forward these messages from an isolated interface */
-  if (in_if->mode == IF_MODE_ISOLATED) {
-    return false;
-  }
-
   /* Forward the message */
   return true;
 }
-- 
2.41.0

